### PR TITLE
📝 Create a combined file with all of the insights from Amazon Comprehend

### PR DESCRIPTION
**Pull request by mermop**
- Create a script to combine text files with the insights from Amazon Comprehend - `zero-carbon-bill/combine.py` - mostly using the code from the `aws-output.ipnyb` notebook
- Combine all of the Amazon Comprehend insights in a .json file for each analysed excerpt
- Remove references to one problematic submission (I'll give you more details on this one in private)
- Move files from the messy `.pdf.txt.json` filenames to just `.json` (this will be most of the diff, sorry)

This will allow us to create a front-end to explore the insights and assess them

### CODE REVIEW

File: aws-output.ipynb [hunk 1]
```diff
@@ -217,7 +217,7 @@
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "zero-carbon-bill/input/09511_John_Adams_Redacted.pdf.txt.json\n",
+      "zero-carbon-bill/input/09511_John_Adams_Redacted.json\n",
```
Comment (Br3nda) [inline aws-output.ipynb:220]: :+1:

File: zero-carbon-bill/combine.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+from glob import glob
+import csv
+
+def sort_by_line_number(elem):
+    return elem['line_number']
+
+def sort_by_weight(elem):
+    return elem['weight']
+
+input_file_pattern = 'input/*.json'
```
Comment (Br3nda) [inline zero-carbon-bill/combine.py]: can you wrap this bit in 
```python
if __name__ == "__main__":
```

File: zero-carbon-bill/combine.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            next
+        else:
+            filename = row[0]
+            item = {'topic': row[1], 'confidence': row[2]}
+
+            if docs_with_topics.get(filename, None) is None:
+                docs_with_topics[filename] = []
+            docs_with_topics[filename].append(item)
+
+def key_phrases_for_submission(filename):
```
Comment (Br3nda) [inline zero-carbon-bill/combine.py]: how about moving the functions to the top of the file

File: zero-carbon-bill/combine.py [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+for key, value in topic_terms.items():
+    topic_terms[key] = sorted(value, key=sort_by_weight, reverse=True)
+
+
+# Import topic modelling: documents
+
+docs_with_topics = {}
+
+with open('topic-modelling/doc-topics.csv', newline='') as doc_topics:
```
Comment (Br3nda) [inline zero-carbon-bill/combine.py]: this "topic-modelling/doc-topics.csv" could be in a variable set at the top of the file.

File: zero-carbon-bill/combine.py [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+def sort_by_line_number(elem):
+    return elem['line_number']
+
+def sort_by_weight(elem):
+    return elem['weight']
+
+input_file_pattern = 'input/*.json'
+input_folder_name = 'input/'
+output_folder_name = 'combined/'
```
Comment (Br3nda) [inline zero-carbon-bill/combine.py]: These are treated as (file) global constants, so it's good practise to make them upper case

File: zero-carbon-bill/combine.py [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+    data = everything_for_file(filename)
+    destination = filename.replace(input_folder_name, output_folder_name)
+    with open(destination, 'w') as outfile:
+        json.dump(data, outfile)
+
+with open(output_folder_name + 'topics.json', 'w') as outfile:
+    json.dump(topic_terms, outfile)
+
+with open(output_folder_name + 'meta.json', 'w') as outfile:
```
Comment (Br3nda) [inline zero-carbon-bill/combine.py]:
```suggestion
with open("{folder}/{file}.json'.format(folder=output_folder_name, file='meta'), 'w') as outfile:
```

File: zero-carbon-bill/combine.py [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+        input_file_name = input_folder_name + filename
+
+    sentiment = sentiment_analysis[filename]
+
+    with open(input_file_name) as f:
+        submission_contents = f.readlines()
+
+    labelled_line_numbers = []
+
+    for l in sentiment:
```
Comment (Br3nda) [inline zero-carbon-bill/combine.py]: this whole data structure is sparkly and useful :sparkler:

File: zero-carbon-bill/combine.py (hunk 2 above)
Comment (mermop) [reply zero-carbon-bill/combine.py]: Which bit? This line or the whole thing? What does that do?

File: zero-carbon-bill/combine.py (hunk 4 above)
Comment (mermop) [reply zero-carbon-bill/combine.py]: done in 999d2ad

File: zero-carbon-bill/combine.py (hunk 3 above)
Comment (mermop) [reply zero-carbon-bill/combine.py]: done in 999d2ad

**Pull request merged**